Sources/SwiftFormat/SwiftFormatter.swift

```diff
@@ -54,8 +54,7 @@ public final class SwiftFormatter {
   /// - Throws: If an unrecoverable error occurs when formatting the code.
   public func format<Output: TextOutputStream>(
     contentsOf url: URL,
-    to outputStream: inout Output,
-    parsingDiagnosticHandler: ((Diagnostic) -> Void)? = nil
```

Comment:
I have mixed feelings about dropping this entirely. For the overload that takes a `SourceFileSyntax`, it makes sense to assume that the user has run the diagnostic-generating visitor over the tree first if they're interested in those. But the overloads that take source text or a URL are a nice convenience and I don't think we want them to silently drop diagnostics.

One idea I had was to change the callback to take the syntax tree instead and the caller could get the diagnostics that way, but by that point it's more natural for the user to just parse the file themselves and run the visitor. What I was doing in my branch was to update the callback to take the `Diagnostic` and `SourceLocation` and then run the visitor here to send that information back.

WDYT?

Reply:
I agree, this is definitely a convenient spot to run the diagnostic post-pass. I'll try to restore this.